lib/Hakyll/Core/Provider.hs

```diff
@@ -24,20 +25,29 @@ module Hakyll.Core.Provider
 
 
 --------------------------------------------------------------------------------
+import           Hakyll.Core.Metadata
 import qualified Hakyll.Core.Provider.Internal      as Internal
 import qualified Hakyll.Core.Provider.MetadataCache as Internal
 import           Hakyll.Core.Store                  (Store)
 
 
 --------------------------------------------------------------------------------
--- | Create a resource provider
+-- | Create a resource provider with the void metadata provider
 newProvider :: Store                   -- ^ Store to use
             -> (FilePath -> IO Bool)   -- ^ Should we ignore this file?
             -> FilePath                -- ^ Search directory
             -> IO Internal.Provider    -- ^ Resulting provider
-newProvider store ignore directory = do
+newProvider store ignore = newProvider' store ignore (const $ return mempty)
+
+-- | Create a resource provider
+newProvider' :: Store                   -- ^ Store to use
```

Comment:
Even though it's a bunch of boilerplate, we should probably add a new `ProviderOptions` together with `defaultProviderOptions` so we can add new fields in a backwards-compatible way.